sdks/java/testing/jpms-tests/build.gradle

```diff
@@ -21,7 +21,14 @@ import groovy.json.JsonOutput
 plugins {
   id 'org.apache.beam.module'
 }
-javaVersion="1.11"
+
+// overwrite javaVersion before applyJavaNature
+if (project.hasProperty("compileAndRunTestsWithJava17")) {
+  javaVersion = '1.17'
+} else {
+  javaVersion = '1.11'
+}
+
 applyJavaNature(
   exportJavadoc: false,
```

Comment:
Note that jpms-tests is a test module that does not publish to maven, so this PR won't change RC